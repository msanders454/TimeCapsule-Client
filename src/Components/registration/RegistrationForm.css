form{
    margin-left:16%;
}

.full_name, .user_name, .password{
    color: white;
    font-weight: 200;
    padding-top: 10px;
}

.registration{
    font-size: 30px;
    text-align: center;
    color: white;
}


label {
    font-size: 15px;
    font-weight: 700;
    display: block;
}

.submit {
    margin: 20px 10px 10px;
    margin-left:20%;
    padding: 10px;
    font-size: 1.3rem;
    letter-spacing: 2px;
    border-width: 10px;
    border-radius: 30px;
    border-color: rgb(8, 6, 56);
    background: rgb(245, 113, 5);
    font-family: 'DM Mono', monospace;
    text-transform: uppercase;
    cursor: pointer;
    color: #fff;
}

.submit:hover {
    background-color: rgb(230, 89, 54);
  }

  @media only screen and (min-width:400px){
    .submit {
        margin-left:27%;
    }
  }

  @media only screen and (min-width:760px){
    .submit {
        margin-left:35%;
    }
  }