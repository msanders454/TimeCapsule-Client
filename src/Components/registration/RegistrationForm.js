/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import { Button2, Required, Form, Input2 } from '../../Utilities/Utils';
import AuthApiService from '../../apiServices/auth-api-service';
import './RegistrationForm.css';

/*
* Used to accept user input and check Autherization status to create a new account. 
*/ 
export default class RegistrationForm extends Component {
    constructor(props) {
        super(props)
        this.state = { error: null };
    }
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    handleSubmit = ev => {
        ev.preventDefault();
        this.setState({ error: null });
        const { full_name, user_name, password } = ev.target;
        AuthApiService.postUser({
            full_name: full_name.value,
            user_name: user_name.value,
            password: password.value,
        })
            .then(user => {
                full_name.value = '';
                user_name.value = '';
                password.value = '';
                this.props.onRegistrationSuccess()
                this.props.history.push('/login')
                console.log(this.props)
            })
                .catch(res => {
                    this.setState({ error: res.error })
                })
    };

    render() {
        const { error } = this.state;
        return (
            <Form className='RegistrationForm' onSubmit={this.handleSubmit}>
                <div role='alert' className= 'alert'>{error && <p className='red'>{error}</p>}</div>
                <div className='full_name'>
                    <h2 className = 'registration'>Register Here</h2>
                    <hr />
                    <label htmlFor='RegistrationForm__full_name'>
                        Full Name <Required />
                    </label>
                    <Input2
                        name='full_name'
                        type='text'
                        required
                        id='fullname'
                        autoComplete='off'
                    ></Input2>
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm__user_name'>
                        Username <Required />
                    </label>
                    <Input2
                        name='user_name'
                        type='text'
                        required
                        id='user_name'
                        autoComplete='off'
                    ></Input2>
                </div>
                <div className='password'>
                    <label htmlFor='RegistrationForm__password'>
                        Password <Required />
                    </label>
                    <p className='Criteria'>(Must have 8 characters with one upper-case letter, one lower-case letter, one number, an one special character)</p>
                    <Input2
                        className='RegistrationForm__Password__Input'
                        name='password'
                        type='password'
                        required
                        id='password'
                        autoComplete='off'
                    ></Input2>
                </div>
                <Button2 type='submit' className='submit'>Register</Button2>
            </Form>
        );
    }
}