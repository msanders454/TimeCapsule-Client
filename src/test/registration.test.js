/* eslint-disable no-undef */
import React from "react";
import { shallow, configure } from "enzyme";
import RegistrationForm from "../Components/registration/RegistrationForm";
import Adapter from "enzyme-adapter-react-16";
 
configure({ adapter: new Adapter() });
test("should render RegistrationForm correctly", () => {
  const wrapper = shallow(<RegistrationForm />);
  expect(wrapper).toMatchSnapshot();
});