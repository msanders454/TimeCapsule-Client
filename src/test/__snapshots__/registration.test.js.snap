// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegistrationForm correctly 1`] = `ShallowWrapper {}`;
